packages/tpetra/core/src/Tpetra_ConfigDefs.hpp

```diff
@@ -129,9 +129,19 @@ namespace Tpetra {
   /*! Allocation profile for matrix/graph entries */
   enum ProfileType {
     StaticProfile,  /*!< Single, static allocation (strict and more efficient) */
+#ifdef TPETRA_ENABLE_DEPRECATED_CODE
     DynamicProfile  /*!< Multiple, dynamic allocations (flexibile, but less efficient) */
+#else
+    DynamicProfile TPETRA_DEPRECATED
+#endif
   };
 
+#ifdef TPETRA_ENABLE_DEPRECATED_CODE
```

Comment:
Do we want to expose this to users?  I'm thinking no -- users either get the default, or they specify the value explicitly.  If we don't want to expose this to users, then we should make this a constant in the `Tpetra::Details` namespace, not a macro.  If we _do_ want to expose it to users, then we should make it a constant in the `Tpetra` namespace, not a macro.

In general, prefer constants and other code things to macro / preprocessor things.

Reply:
You're right, this code doesn't even do anything.  I had meant to cull it previously, but thanks for catching it.